###Forcasting Accuracy
Measures:<br/>
+ Mean Squared Error (MSE): Measures the average of the squared differences between what you predicted and what actually happened.
+ Mean Absolute Error (MAE): Measures the average of the absolute difference between your prediction and the actual values.
+ Mean Absolute Percentage Error (MAPE): Measures the average of the absolute percantage differences between your predictions and the actual values.

In [1]:
import pyodide_http
pyodide_http.patch_all()
import warnings
warnings.filterwarnings('ignore')
import micropip
await micropip.install('pandas')
await micropip.install('openpyxl')
import pandas as pd

# file with data
excel_file = pd.ExcelFile("forecast_errors.xlsx", engine='openpyxl')

data = pd.read_excel(excel_file)
data.head()

# Calculate MSE
mse = ((data['Actual_Price'] - data['Forecast_Price']) ** 2).mean()
# Print MSE
print(f'The Mean Squared Error (MSE) is: {mse}')

# Calculate MAE
mae = (data['Actual_Price'] - data['Forecast_Price']).abs().mean()
# Print MAE
print(f'The Mean Absolute Error (MAE) is: {mae}')

# Calculate MAPE
mape = (abs(data['Actual_Price'] - data['Forecast_Price'])/data['Actual_Price']).mean()*100
# Print MAPE
print(f'Mean Absolute Percentage Error (MAPE) is: {mape}')

The Mean Squared Error (MSE) is: 1.6
The Mean Absolute Error (MAE) is: 1.2
Mean Absolute Percentage Error (MAPE) is: 2.2805959938035407


**The Mean Squared Error (MSE) is: 1.6**<br/>
In summary, on average, the squared errors (different between actual and forecasted prices) are 1.6. Since the errors are squared, this value emphasizes larger errors more than smaller ones. A lower value indicates more accurate forecasts, as the difference between actual and forecasted values are smaller.<br/>
**The Mean Absolute Error (MAE) is: 1.2**<br/>
The MAE value of 1.2 suggests that the forecasts are relatively close to the actual prices, with an average error of 1.2 dollars.<br/>
**Mean Absolute Percentage Error (MAPE) is: 2.2805959938035407**
On average, the forecasted stock prices differ from the actual stock prices by 2.3%. This indicates that the predicted stock prices are very close to the actual prices, making the forecast reliable for decision-making.

In [3]:
# file with data
excel_file = pd.ExcelFile("fans_data.xlsx", engine='openpyxl')

data = pd.read_excel(excel_file)
data.head()

# Calculate MSE
mse = ((data['Actual Sales'] - data['Forecasted Sales']) ** 2).mean()
# Print MSE
print(f'The Mean Squared Error (MSE) is: {mse}')

# Calculate MAE
mae = (data['Actual Sales'] - data['Forecasted Sales']).abs().mean()
# Print MAE
print(f'The Mean Absolute Error (MAE) is: {mae}')

# Calculate MAPE
mape = (abs(data['Actual Sales'] - data['Forecasted Sales'])/data['Actual Sales']).mean()*100
# Print MAPE
print(f'Mean Absolute Percentage Error (MAPE) is: {mape}')

The Mean Squared Error (MSE) is: 1450.0
The Mean Absolute Error (MAE) is: 35.0
Mean Absolute Percentage Error (MAPE) is: 5.199862637362638
